opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -851,7 +851,7 @@ namespace Opm
 
         // only use inner well iterations for the first newton iterations.
         const int iteration_idx = simulator.model().newtonMethod().numIterations();
-        if (iteration_idx < this->param_.max_niter_inner_well_iter_ || this->well_ecl_.isMultiSegment()) {
```

Comment:
If we go this direction, we probably want to do it differently here. 
`iteration_idx` by default will be maximum 20. We probably can remove this if condition or using some other if condition if I did not read the code wrong.

Reply:
I think it could be useful to keep the option of not solving with inner iterations for all Newton iterations.